server/src/logger.ts

```diff
@@ -132,15 +125,15 @@ export class Logger implements ts.server.Logger {
   }
 
   loggingEnabled() {
-    return !!this.logFilename || this.traceToConsole;
+    return this.fd >= 0;
   }
 
   hasLevel(level: ts.server.LogLevel) {
     return this.loggingEnabled() && this.level >= level;
   }
 
   msg(s: string, type: ts.server.Msg = ts.server.Msg.Err) {
-    if (!this.canWrite) return;
+    if (!this.loggingEnabled()) return;
```

Comment:
Check is unnecessary since `write` checks for this already

Reply:
yeah, but with this check we can be slightly more efficient by not processing the string unnecessarily

Reply:
Can we just move the contents of `write` into here then, as it's the only caller

Reply:
done